server/migrations/0043_tournament_match_names.py

```diff
@@ -0,0 +1,112 @@
+    else:
+        PositionPool = apps.get_model("server", "PositionPool")  # noqa: N806
+        pools = PositionPool.objects.all()
+
+    for pool in pools:
+        pool_seeding_list = sorted(map(int, pool.initial_seeding.keys()))
+
+        for i, seed_x in enumerate(pool_seeding_list):
+            for j, seed_y in enumerate(pool_seeding_list[i + 1 :], i + 1):
+                pool_match_name = f"{pool.name}{i + 1} vs {pool.name}{j + 1}"
+
+                Match = apps.get_model("server", "Match")  # noqa: N806
+
+                pool_match = Match.objects.get(
+                    tournament=pool.tournament,
+                    pool=pool if is_pool else None,
+                    position_pool=pool if (not is_pool) else None,
+                    placeholder_seed_1=seed_x,
+                    placeholder_seed_2=seed_y,
+                )
+
+                pool_match.name = pool_match_name
+                pool_match.save()
+
+
+def simulate_creating_bracket_sequence_matches(  # type: ignore[no-untyped-def]
+    apps: StateApps,
+    bracket,  # noqa: ANN001
+    start: int,
+    end: int,
+    seq_num: int,
+) -> None:
+    for i in range(0, ((end - start) + 1) // 2):
+        seed_1 = start + i
+        seed_2 = end - i
+
+        bracket_match_name = get_bracket_match_name(start, end, seed_1, seed_2)
+
+        Match = apps.get_model("server", "Match")  # noqa: N806
+        bracket_match = Match.objects.get(
+            tournament=bracket.tournament,
+            bracket=bracket,
+            placeholder_seed_1=seed_1,
+            placeholder_seed_2=seed_2,
+        )
+
+        bracket_match.name = bracket_match_name
+        bracket_match.sequence_number = seq_num
+
+        bracket_match.save()
+
+    if end - start > 1:
+        simulate_creating_bracket_sequence_matches(
+            apps,
+            bracket,
+            start,
+            start + (((end - start) + 1) // 2) - 1,
+            seq_num + 1,
+        )
+        simulate_creating_bracket_sequence_matches(
+            apps, bracket, start + (((end - start) + 1) // 2), end, seq_num + 1
+        )
+
+
+def simulate_creating_bracket_matches(  # type: ignore[no-untyped-def]
+    apps: StateApps, bracket  # noqa: ANN001
+) -> None:
+    start, end = map(int, bracket.name.split("-"))
+    if ((end - start) + 1) % 2 == 0:
+        simulate_creating_bracket_sequence_matches(apps, bracket, start, end, 1)
+
+
+def generate_bracket_match_names(apps: StateApps) -> None:
+    Bracket = apps.get_model("server", "Bracket")  # noqa: N806
+    brackets = Bracket.objects.all()
+    for bracket in brackets:
+        simulate_creating_bracket_matches(apps, bracket=bracket)
+
+
+def generate_match_names(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    """Generate match names for pools, position pools and brackets"""
+    generate_pool_match_names(apps, is_pool=True)
+    generate_pool_match_names(apps, is_pool=False)
+    generate_bracket_match_names(apps)
+
```

Comment:
Hmm what about cross pool match names?

Reply:
for now, we decided to generate names for pools and brackets alone right ? do we have any ideas for cross pool match names ?

Reply:
I thought we agreed to just display "Cross Pool" / CP in small devices right?

Reply:
So we can save that in the name?

Reply:
sure, we can do that. I'll add "Cross Pool" as the name